app/javascript/components/public_health/query/JurisdictionFilter.js

```diff
@@ -1,19 +1,28 @@
 import React from 'react';
 import { PropTypes } from 'prop-types';
 import { Button, Form, InputGroup, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import _ from 'lodash';
 
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 class JurisdictionFilter extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      sorted_jurisdiction_paths: [],
       jurisdiction_path: props.jurisdiction_paths[props.jurisdiction] || '',
     };
   }
 
   handleJurisdictionChange = jurisdiction_path => {
-    this.setState({ jurisdiction_path }, () => {
+    // Populate & sort the jurisdiction paths if it hasn't been populated yet
```

Comment:
Could this be done in a more React-y way? Maybe when the component gets its props updated, setting it that way? maybe something like `getDerivedStateFromProps`?

Reply:
Thanks for the suggestion. Just pushed an update with it.